### DISCUSSION TITLE
IES Lights in Filament

### DISCUSSION THREAD

**Discussion by Nolram12345** [google/filament#7783] [Q&A] [answered]
IES Lights are mentioned in the [Filament PBR document](https://google.github.io/filament/Filament.md.html#lighting/directlighting/photometriclights), but I can't seem to actually find these referenced in any Filament API. There are references to an IES color in the punctual light shader, but I can't quite figure out how to actually use them.

Are IES lights currently implemented in Filament, and if so, how are they used?

**Comment by romainguy** [comment, marked as answer]
IES lights are currently not implemented. They were part of our initial prototype though.

**Reply by Nolram12345** [reply]
Thank you for the answer! I would presume this is something that may appear in the future? Would alow for adding some additional detail to lights without performance implications - but I am aware ya'll are very busy, so I understand it's certainly not a high priority.

**Comment by Rizhiy825** [comment]
Hi! Any updates here?
We really need IES profiles in out apps.

Does it make sense for me to do this and contribute IES functionalty?

**Reply by romainguy** [reply]
@pixelflinger should answer this but I would just like to point out that the limited number of texture units on some of the target platforms/devices (Android) has been a major hurdle to adding this feature. IES profiles can however be implemented as 1D buffers (which puts limitations on what IES profiles can be recreated, but it covers a large number of them). In this case since an IES light would just take 256 bytes, maybe a UBO would be a good place where to store the IES profiles? Another issue is whether there's room in the lights UBO for an extra index. Maybe just borrowing a few bits and putting cap on how many concurrent profiles can be used?

**Reply by Rizhiy825** [reply]
Thanks for your answer.
I agree with you and think that UBO is a suitable place to store IES data.

We can choose the max number of different IES, for example 256.
In my experience with Unity engine, this is enought for many scenes.

It would be great to work with IES!
